Google.PowerShell/Compute/GceCmdlet.cs

```diff
@@ -110,6 +122,51 @@ private void WriteWarnings(Operation op)
             }
         }
 
+        private void WriteProgress(Operation op, string progressMessage)
+        {
+            int activityId = (int)(op.Id % int.MaxValue ?? 0);
+            string activity = progressMessage ?? op.Description;
+            if (activity == null)
+            {
+                string operationType = op.OperationType;
+                operationType = operationType.First().ToString().ToUpper() + operationType.Substring(1);
+                int startIndex = op.TargetLink.IndexOf("projects", StringComparison.Ordinal);
+                if (startIndex < 0 || startIndex >= op.TargetLink.Length)
+                {
+                    startIndex = 0;
+                }
+                activity = operationType + " " + op.TargetLink.Substring(startIndex);
+            }
+            string statusDescription = op.StatusMessage ?? op.Status;
+            int percentComplete;
+            if (op.Progress == null || op.Progress == 0)
+            {
+                percentComplete = _spinnerValue = (_spinnerValue + 1) % 100;
+            }
+            else
+            {
+                percentComplete = (int)op.Progress;
+            }
+            ProgressRecord record = new ProgressRecord(activityId, activity, statusDescription)
+            {
+                RecordType = ProgressRecordType.Processing,
+                PercentComplete = percentComplete,
+            };
+            WriteProgress(record);
+        }
+
+        private void EndProgress(Operation op)
+        {
+            int activityId = (int)(op.Id % int.MaxValue ?? 0);
```

Comment:
1. We need to explicitly initialize `_spinnerValue` to 0. Yes, the CLR spec says it will default to 0, but we are better than that.
2. When a progress bar ends, we need to reset the value of `_spinnerValue` to 0.